main/src/ca/uwaterloo/flix/tools/pkg/ManifestParser.scala

```diff
@@ -16,22 +16,240 @@
 package ca.uwaterloo.flix.tools.pkg
 
 import ca.uwaterloo.flix.util.Result
-import org.tomlj.Toml
+import ca.uwaterloo.flix.util.Result.{Err, Ok}
+import org.tomlj.{Toml, TomlArray, TomlInvalidTypeException, TomlTable}
 
-import java.io.{PrintStream, StringReader}
-import java.nio.file.Path
+import java.io.{IOException, PrintStream}
+import java.nio.file.{InvalidPathException, Path, Paths}
+import scala.collection.mutable
 
 object ManifestParser {
 
-  def parse(p: Path)(implicit out: PrintStream): Result[Manifest, ManifestError] = ???
+  def parse(p: Path)(implicit out: PrintStream): Result[Manifest, ManifestError] = {
+    val parser = try {
+      val parser = Toml.parse(p)
+      parser
+    } catch {
+      case _: IOException => return Err(ManifestError.IOError(p))
+    }
+
+    val errors = parser.errors
+    if(errors.size() > 0) {
+      var error_string = ""
+      errors.forEach(error => error_string = error_string + error.toString + ", ")
+      return Err(ManifestError.ManifestParseError(p, error_string))
+    }
+
+    val name = try {
```

Comment:
Would it make sense to extract e.g. lines 43 to 51 into one helper function? Like:

```
def getRequiredStringProperty(p: String, toml: ...): Result[String, ManifestError]
```